### PR TITLE
Fixed Breaking Bar Height

### LINKED ISSUE #851: Blank Circuit Name Breaks Bar Height
image

When names of all project circuits are empty the circuit selection bar changes width to 5.9px. The best solution will be to don't allow for blank names
_This issue is a part of my [Usability Test](https://drive.google.com/open?id=1igwLx2tc1dAU_75qY2F4u7ZnHf2crcp-)_

**Are you working on this issue? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by Shivansh2407**
Fixes #851 Blank Circuit Name Breaks Bar Height 

#### Describe the changes you have made in this pr -
I have Now made the field of `Circuit :` as `required` . Also If during the change of Name of circuit the Circuit name is blank (for a short span of time) then for that case I have fixed the tabsBar using another span in the same tabsBar. This will Prevent the TabsBar to collapse

### Screenshots of the changes (If any) - 
Cap1
Cap2

### CODE REVIEW

**Comment by Shivansh2407** [comment]
Please Review this @satu0king and @Ananthavijay . I will be squashing the commits soon.

**Comment by Shivansh2407** [comment]
@tachyons Please Review this PR and Merge 👍

**Comment by Shivansh2407** [comment]
Fixed The repeating tag of Circuit Name : . Now only circuit name will appear once rather than repeated times on adding new circuits.
Please review @satu0king 👍

**Comment by Shivansh2407** [comment]
> I agree on removing the Circuit Name label or maybe making it look better, also attach a video recording of the solved issue as well.

Will Update this soon @nitin10s

**Comment by Shivansh2407** [comment]
I have updated the PR with requested changes . Please review it @nitin10s and @tachyons 👍 .
patch-4 updated

**Comment by Shivansh2407** [comment]
Ok @MrBartusek will fix it soon . @tachyons and @nitin10s please review 👍

**Comment by Shivansh2407** [comment]
@tachyons and @nitin10s  please review 👍

**Comment by Shivansh2407** [comment]
@satu0king and @nitin10s please review this  👍

File: public/js/data.js [hunk 1]
```diff
@@ -25,6 +25,9 @@ function newCircuit(name, id) {
 
 
 function changeCircuitName(name, id = globalScope.id) {
+      if(name == "") {
+        name = "Empty";
+      }
```
Comment (satu0king) [inline public/js/data.js]: replace with 

name = name || "Untitled"

Comment (Shivansh2407) [reply public/js/data.js]: It is working 👍 @satu0king . Please Review and Merge

**Comment by Shivansh2407** [comment]
#1295 Opened with a single change. Closing this PR.

**Pull request closed without merging**